server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/AESPayloadCodec.java

```diff
@@ -71,7 +89,9 @@ public Payload write(InputStream is) {
     @Override
     public InputStream read(Payload payload) throws IOException {
         try {
-            return streamingAead.newDecryptingStream(payload.openStream(), PBKDF2StreamingAeadFactory.EMPTY_ASSOCIATED_DATA);
+            byte[] ciphertext = IOUtils.toByteArray(payload.getPayload().openStream());
+            byte[] decrypt = aead.decrypt(ciphertext, EMPTY_ASSOCIATED_DATA);
+            return new ByteArrayInputStream(decrypt);
```

Comment:
you are encrypting with a FileBackedOutputStream but decrypting with a ByteArrayInputStream?

Reply:
Yes, there's on `FileBackedInputStream` implementation & you have to read the whole stream in order to decrypt it.